### ISSUE TITLE
Requests backend for HTTP fetches is broken after removing streamed response handlers

### ISSUE THREAD

**Issue by The-Loeki** [saltstack/salt#27794]
Commit acf2d51440af97836b48adb18276df5263590f51 modified `salt/fileclient.py` to directly write out the `.body` of the response rather than a streaming handler.

This is all well & good, but the `requests` backend result does not contain a `.body`, resulting in 
```
[ERROR   ] An exception occurred in this state: Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/salt/state.py", line 1590, in call
    **cdata['kwargs'])
  File "/usr/lib/python2.7/dist-packages/salt/loader.py", line 1483, in wrapper
    return f(*args, **kwargs)
  File "/usr/lib/python2.7/dist-packages/salt/states/pkg.py", line 1081, in installed
    **kwargs)
  File "/usr/lib/python2.7/dist-packages/salt/modules/aptpkg.py", line 520, in install
    name, pkgs, sources, **kwargs
  File "/usr/lib/python2.7/dist-packages/salt/modules/pkg_resource.py", line 140, in parse_targets
    srcinfo.append(__salt__['cp.cache_file'](pkg_src, saltenv))
  File "/usr/lib/python2.7/dist-packages/salt/modules/cp.py", line 365, in cache_file
    result = __context__['cp.fileclient'].cache_file(path, saltenv)
  File "/usr/lib/python2.7/dist-packages/salt/fileclient.py", line 155, in cache_file
    return self.get_url(path, '', True, saltenv)
  File "/usr/lib/python2.7/dist-packages/salt/fileclient.py", line 614, in get_url
    destfp.write(query['handle'].body)
AttributeError: 'Response' object has no attribute 'body'
```

The result is actually in `.content` for a `requests` Response.

The problem is rather pressing for us because of #8177/#23617/#21985. Currently the *only* way to get Salt's HTTP requests to use a proxy is by utilizing the `requests` backend & specifiying the environment variables.

**Comment by jfindlay** [comment]
@The-Loeki, thanks for the report.

@jacksontj

**Comment by jacksontj** [comment]
IIRC that fix was pushed to 2015.8, PR is #28066 for the fix

**Comment by The-Loeki** [comment]
I'm afraid the issue is far from over; I've commented in #28065
